src/components/combat.rs

```diff
@@ -0,0 +1,197 @@
+use amethyst::core::Named;
+use amethyst::ecs::{
+    Component, DenseVecStorage, Entity, HashMapStorage, LazyUpdate, Read, ReadStorage, VecStorage,
+};
+use amethyst::prelude::*;
+use amethyst_imgui::imgui;
+use std::time::Duration;
+
+pub struct Damage {
+    // Points subtracted from target's health per hit
+    pub damage: f32,
+}
+
+impl Component for Damage {
+    type Storage = DenseVecStorage<Self>;
+}
+
+impl Damage {
+    pub fn new(damage: f32) -> Damage {
+        Damage { damage }
+    }
+}
+
+impl<'a> amethyst_inspector::Inspect<'a> for Damage {
+    type SystemData = (ReadStorage<'a, Self>,);
+
+    fn inspect((storage,): &Self::SystemData, entity: amethyst::ecs::Entity, ui: &imgui::Ui<'_>) {
+        let &Damage { mut damage } = if let Some(x) = storage.get(entity) {
+            x
+        } else {
+            return;
+        };
+        ui.drag_float(imgui::im_str!("value##damage{:?}", entity), &mut damage)
+            .build();
+        ui.separator();
+    }
+}
+
+///
+///
+///
+pub struct Speed {
+    pub attacks_per_second: f32,
+}
+
+impl Component for Speed {
+    type Storage = DenseVecStorage<Self>;
+}
+
+impl Speed {
+    pub fn new(attacks_per_second: f32) -> Speed {
+        Speed { attacks_per_second }
+    }
+}
+
+impl<'a> amethyst_inspector::Inspect<'a> for Speed {
+    type SystemData = (ReadStorage<'a, Self>,);
+
+    fn inspect((storage,): &Self::SystemData, entity: amethyst::ecs::Entity, ui: &imgui::Ui<'_>) {
+        let &Speed {
+            mut attacks_per_second,
+        } = if let Some(x) = storage.get(entity) {
+            x
+        } else {
+            return;
+        };
+        ui.drag_float(
+            imgui::im_str!("attacks per second##speed{:?}", entity),
+            &mut attacks_per_second,
+        )
+        .build();
+        ui.separator();
+    }
+}
+
+///
+///
+///
+// As long as the cooldown component is attached to an entity, that entity won't be able to attack.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Cooldown {
```

Comment:
Any ideas for alternative implementations of the cooldown? What we want to achieve is that and entity can only attack `x` times a second.